app/controllers/application_controller.rb

```diff
@@ -430,7 +430,16 @@ def use_caching?
   def check_user_status
     if current_user.is_a?(User) && (current_user.suspended? || current_user.banned?)
       if current_user.suspended?
-        flash[:error] = t('suspension_notice', default: "Your account has been suspended until %{suspended_until}. You may not add or edit content until your suspension has been resolved. Please <a href=\"#{new_abuse_report_path}\">contact Abuse</a> for more information.", suspended_until: localize(current_user.suspended_until)).html_safe
+        suspension_end = current_user.suspended_until
+
+        # Unban threshold is 6:51pm, 12 hours after the unsuspend_users rake task located in schedule.rb is run at 6:51am
+        unban_theshold = DateTime.new(suspension_end.year, suspension_end.month, suspension_end.day, 18, 51, 0, "+00:00") 
+
+        # If the stated suspension end date is after the unban threshold we need to advance a day 
+        suspension_end = suspension_end.next_day(1) if suspension_end > unban_theshold
+        localized_suspension_end = DateTime.parse(localize(suspension_end)).strftime("%Y-%m-%d")
+        flash[:error] = t("suspension_notice", default: "Your account has been suspended until %{suspended_until}. You may not add or edit content until your suspension has been resolved. Please <a href=\"#{new_abuse_report_path}\">contact Abuse</a> for more information.", suspended_until: localized_suspension_end).html_safe
```

Comment:
I've edited line 441 to be like so 

```
flash[:error] = t(".users.status.suspension_notice", suspended_until: localized_suspension_end, contact_abuse_link: view_context.link_to(t(".contact_abuse"), new_abuse_report_path)).html_safe
```

and added the following to en.yml:

```
status:
      ban_notice: Your account has been banned. You are not permitted to add or edit archive content. Please %{contact_abuse_link} for more information.
      suspension_notice: Your account has been suspended until %{suspended_until}. You may not add or edit content until your suspension has been resolved. Please %{contact_abuse_link} for more information.
```

Reply:
However I still get missing translations and unused key errors, so something is wrong with the link between the controller and the translation file. Do you know what I'm doing wrong?

Reply:
It looks like the test is expecting the following absolute/fully-qualified keys
```
+--------+----------------------------------------------------+--------------------------------------------------------+
| Locale | Key                                                | Value in other locales or source                       |
+--------+----------------------------------------------------+--------------------------------------------------------+
|   en   | application.check_user_not_suspended.contact_abuse | app/controllers/application_controller.rb:423          |
|   en   | application.check_user_status.contact_abuse        | app/controllers/application_controller.rb:401 (1 more) |
+--------+----------------------------------------------------+--------------------------------------------------------+
```

If you wanted to use a key like https://github.com/otwcode/otwarchive/blob/ea83a19650d958113e61fac73a9d2882e5d99a29/config/locales/controllers/en.yml#L93, it would need to be absolute (`users.passwords.create.contact_abuse`) -- although I would recommend extracting that key to a common location if it's used in multiple controllers